### PR TITLE
Fix/382 boto3

### PR DESCRIPTION
**Pull request by arthurHamon2**
This is a duplicated PR of this one:  https://github.com/kevin1024/vcrpy/pull/411
The tests are fixed on this PR, enabling to (finally) merge the boto3 final integration.

### CODE REVIEW

**Comment by arthurHamon2** [comment]
@1oglop1 can you review the changes in order to finally merge this PR ?

**Comment by 1oglop1** [comment]
@arthurHamon2 Thanks for working on it, I will look into it this week.

File: .travis.yml [hunk 1]
```diff
@@ -10,6 +10,7 @@ env:
     - TOX_SUFFIX="httplib2"
     - TOX_SUFFIX="boto3"
```
Comment (neozenith) [inline .travis.yml]: Below in the `allowed_failures` there is still `boto3` and travis is failing on this environment. So this doesn't address #382 from what I'm seeing.

File: tests/integration/test_boto3.py [hunk 2]
```diff
@@ -1,15 +1,46 @@
...
+ses = boto3.Session(
+    aws_access_key_id=os.environ['AWS_ACCESS_KEY_ID'],
+    aws_secret_access_key=os.environ['AWS_SECRET_ACCESS_KEY'],
+    aws_session_token=None,
+    region_name=os.environ['AWS_DEFAULT_REGION'],
+    # botocore_session=None,
+    # profile_name=None
+)
+
+IAM_CLIENT = ses.client('iam')
```
Comment (neozenith) [inline tests/integration/test_boto3.py]: Would providing the IAM_CLIENT as a fixture be preferred here?

File: tests/integration/test_boto3.py [hunk 3]
```diff
@@ -23,45 +54,59 @@ def test_boto_stubs(tmpdir):
...
+        assert issubclass(VCRRequestsHTTPSConnection, AWSHTTPSConnection)
+        AWSHTTPConnection('hostname.does.not.matter')
+        AWSHTTPSConnection('hostname.does.not.matter')
 
-    # retrieve content to check it
-    o = s3_resource.Object(bucket, key).get()
 
-    # decode for python3
-    assert content == o['Body'].read().decode('utf-8')
+def test_boto3_without_vcr():
```
Comment (neozenith) [inline tests/integration/test_boto3.py]: This test could be removed as it is testing AWS more than it is testing vcrpy.

**Comment by arthurHamon2** [comment]
@neozenith I'll try to fix this boto3 integration by the end of the week, stay tuned !

**Comment by arthurHamon2** [comment]
I'm closing this in favor of https://github.com/kevin1024/vcrpy/pull/411 (I have managed to fix the boto3 integration, PR will be ready today !)
Still I'll take your review into account @neozenith !

**Pull request closed without merging**